modules/subscriptionService/src/main/scala/routes/Routes.scala

```diff
@@ -18,14 +19,54 @@ object Routes {
       jsonEncoderOf[F, PostSubscriptions]
     implicit val decoder: EntityDecoder[F, PostSubscriptions] =
       jsonOf[F, PostSubscriptions]
+    implicit val slackCommandRequestBodyEncoder: EntityEncoder[F, SlackCommandRequestBody] =
+      jsonEncoderOf[F, SlackCommandRequestBody]
+    implicit val slackCommandRequestBodyDecoder: EntityDecoder[F, SlackCommandRequestBody] =
+      jsonOf[F, SlackCommandRequestBody]
 
     HttpRoutes.of[F] {
-      case GET -> Root                => Ok("subscription service")
-      case GET -> subscription / user => Ok(MockedResponse.mockedGetUserSubscriptionResponse)
-      case req @ POST -> subscription / user =>
-        req.decode[PostSubscriptions] { message: PostSubscriptions =>
-          Created(message)
-        }
+      case GET -> Root => Ok("subscription service")
+      case GET -> Root / "subscription" / user =>
+        Ok(MockedResponse.mockedGetUserSubscriptionResponse)
+      case req @ POST -> Root / "subscription" / user =>
+        handleRequest[F, PostSubscriptions](req, "application/json")
+      case req @ DELETE -> Root / "subscription" / user =>
+        handleRequest[F, PostSubscriptions](req, "application/json")
+      case req @ POST -> Root / "subscription" / "slack" / command =>
+        req
+          .withHeaders(
+            Headers.of(
+              Header("Content-Type", "application/x-www-form-urlencoded")
+            )
+          )
+          .attemptAs[SlackCommandRequestBody]
+          .foldF(
+            error => BadRequest(s"An error occurred ${error}"),
+            _ => Accepted()
+          )
     }
   }
+
+  private def handleRequest[F[_]: Sync, A <: Subscription](
+      request: Request[F],
+      contentType: String
+  )(implicit
+      entityDecoder: EntityDecoder[F, A],
+      entityEncoder: EntityEncoder[F, A]
+  ): F[Response[F]] = {
+    val dsl = new Http4sDsl[F] {}
+    import dsl._
+    request
+      .withHeaders(Headers.of(Header("Content-Type", contentType)))
+      .attemptAs[A]
+      .foldF(
+        error => BadRequest(s"Bad request ${error}"),
```

Comment:
@SarpongAbasimi is usually not a good practice to leak error details in the HTTP responses. What we usually do is to log the error and return an empty `BadRequest` error (or some generic message, suited for the client).

Reply:
😬 I will sort this out thanks